src/sentry/replays/endpoints/project_replay_viewed_by.py

```
@@ -103,7 +106,32 @@ def post(self, request: Request, project: Project, replay_id: str) -> Response:
         except ValueError:
             return Response(status=404)
 
-        message = viewed_event(project.id, replay_id, request.user.id)
+        # make a query to avoid overwriting the `finished_at` column
+        filter_params = self.get_filter_params(request, project, date_filter_optional=False)
+        finished_at_response = execute_query(
+            query=make_full_aggregation_query(
+                fields=["finished_at"],
+                replay_ids=[replay_id],
+                project_ids=[project.id],
+                period_start=filter_params["start"],
+                period_end=filter_params["end"],
+                request_user_id=request.user.id,
+            ),
+            tenant_id={"organization_id": project.organization.id} if project.organization else {},
+            referrer="replays.endpoints.viewed_by_post",
+        )["data"]
+
+        finished_at_ts = None
+        if finished_at_response:
+            finished_at = finished_at_response[0]["finished_at"]
+            finished_at_ts = datetime.fromisoformat(finished_at).timestamp()
+
+        message = viewed_event(
+            project.id,
+            replay_id,
+            request.user.id,
+            finished_at_ts if finished_at_ts else time.time(),
```

If we can't find the replay let's abort 404 rather than default.